src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -0,0 +1,184 @@
+import org.broadinstitute.dsde.workbench.leonardo.monitor.ClusterMonitorSupervisor.ClusterDeleted
+
+import scala.concurrent.Future
+import scala.util.Failure
+
+object ClusterMonitorActor {
+  /**
+    * Creates a Props object used for creating a {{{ClusterMonitorActor}}}.
+    */
+  def props(cluster: Cluster, monitorConfig: MonitorConfig, gdDAO: DataprocDAO, dbRef: DbReference): Props =
+    Props(new ClusterMonitorActor(cluster, monitorConfig, gdDAO, dbRef))
+
+  // ClusterMonitorActor messages:
+
+  private[monitor] sealed trait ClusterMonitorMessage extends Product with Serializable
+  private[monitor] case object ScheduleMonitorPass extends ClusterMonitorMessage
+  private[monitor] case object QueryForCluster extends ClusterMonitorMessage
+  private[monitor] case class ReadyCluster(publicIP: String) extends ClusterMonitorMessage
+  private[monitor] case class NotReadyCluster(status: ClusterStatus) extends ClusterMonitorMessage
+  private[monitor] case class FailedCluster(errorDetails: ClusterErrorDetails) extends ClusterMonitorMessage
+  private[monitor] case object DeletedCluster extends ClusterMonitorMessage
+  private[monitor] case class ShutdownActor(notifyParentMsg: Option[Any] = None) extends ClusterMonitorMessage
+}
+
+/**
+  * An actor which monitors the status of a Cluster. Periodically queries Google for the cluster status,
+  * and acts appropriately for Running, Deleted, and Failed clusters.
+  * @param cluster the Cluster to monitor
+  * @param monitorConfig monitor configuration properties
+  * @param gdDAO the Google dataproc DAO
+  * @param dbRef the DB reference
+  */
+class ClusterMonitorActor(val cluster: Cluster,
+                          val monitorConfig: MonitorConfig,
+                          val gdDAO: DataprocDAO,
+                          val dbRef: DbReference) extends Actor with LazyLogging {
+  import context._
+
+  override def preStart(): Unit = {
+    super.preStart()
+    scheduleMonitorPass
+  }
+
+  override def receive: Receive = {
+    case ScheduleMonitorPass =>
+      scheduleMonitorPass
+
+    case QueryForCluster =>
+      checkCluster pipeTo self
+
+    case NotReadyCluster(status) =>
+      handleNotReadyCluster(status) pipeTo self
+
+    case ReadyCluster(ip) =>
+      handleReadyCluster(ip) pipeTo self
+
+    case FailedCluster(errorDetails) =>
+      handleFailedCluster(errorDetails) pipeTo self
+
+    case DeletedCluster =>
+      handleDeletedCluster pipeTo self
+
+    case ShutdownActor(notifyParentMsg) =>
+      notifyParentMsg.foreach(msg => parent ! msg)
+      stop(self)
+
+    case Failure(e) =>
+      // An error occurred, let the supervisor handle it
+      logger.error(s"Error occurred monitoring cluster ${cluster.googleProject}/${cluster.clusterName}", e)
+      throw e
+  }
+
+  private def scheduleMonitorPass: Unit = {
+    system.scheduler.scheduleOnce(monitorConfig.pollPeriod, self, QueryForCluster)
```

Comment:
needs 0 - `pollPeriod` jitter on startup and then normal `addJitter` jitter thereafter

Reply:
Right, jitter! Will do